## `nixpkgs-review` result

Generated using [`nixpkgs-review-gha`](https://github.com/Defelo/nixpkgs-review-gha)

Command: `nixpkgs-review pr 484721`
Commit: [`a87d19d3b962d8946a45cfee9373c1996b335530`](https://github.com/NixOS/nixpkgs/commit/a87d19d3b962d8946a45cfee9373c1996b335530) ([subsequent changes](https://github.com/NixOS/nixpkgs/compare/a87d19d3b962d8946a45cfee9373c1996b335530..pull/484721/head))
Merge: [`62bee906f31f54e6d1cfca1246306d3e77613e13`](https://github.com/NixOS/nixpkgs/commit/62bee906f31f54e6d1cfca1246306d3e77613e13)

Logs: https://github.com/magnetophon/nixpkgs-review-gha/actions/runs/21468654280


---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>qdelay</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>qdelay</li>
  </ul>
</details>

---
### `x86_64-darwin` (sandbox = true)
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>qdelay</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `x86_64-darwin`</summary>
<details>
<summary>qdelay</summary>
<pre>    /nix/store/2g5sj60rd069ibv300i35yn5l4x967rm-clang-wrapper-21.1.8/bin/clang   -arch x86_64 -isysroot /nix/store/0b11sl1s441yhz6fxa6h8nr8h05c8w7p-apple-sdk-14.4/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk -mmacosx-version-min=10.13 -fPIE -MD -MT CMakeFiles/cmTC_9d13f.dir/testCCompiler.c.o -MF CMakeFiles/cmTC_9d13f.dir/testCCompiler.c.o.d -o CMakeFiles/cmTC_9d13f.dir/testCCompiler.c.o -c /nix/build/nix-15667-1880892402/source/build/CMakeFiles/CMakeScratch/TryCompile-Zj19LP/testCCompiler.c
    Linking C executable cmTC_9d13f
    /nix/store/1cyc0yckxw16x6mma6jbcp70c7nv8i8i-cmake-4.1.2/bin/cmake -E cmake_link_script CMakeFiles/cmTC_9d13f.dir/link.txt --verbose=1
    ld: library not found for -lX11
    clang: error: linker command failed with exit code 1 (use -v to see invocation)
    /nix/store/2g5sj60rd069ibv300i35yn5l4x967rm-clang-wrapper-21.1.8/bin/clang  -arch x86_64 -isysroot /nix/store/0b11sl1s441yhz6fxa6h8nr8h05c8w7p-apple-sdk-14.4/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk -mmacosx-version-min=10.13 -Wl,-search_paths_first -Wl,-headerpad_max_install_names CMakeFiles/cmTC_9d13f.dir/testCCompiler.c.o -o cmTC_9d13f
    make[1]: *** [CMakeFiles/cmTC_9d13f.dir/build.make:103: cmTC_9d13f] Error 1
    make[1]: Leaving directory &#x27;/nix/build/nix-15667-1880892402/source/build/CMakeFiles/CMakeScratch/TryCompile-Zj19LP&#x27;
    make: *** [Makefile:134: cmTC_9d13f/fast] Error 2
    
    



  CMake will not be able to correctly generate this project.
Call Stack (most recent call first):
  CMakeLists.txt:11 (project)


-- Configuring incomplete, errors occurred!</pre>
</details>
</details>

---
### `aarch64-darwin` (sandbox = true)
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>qdelay</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `aarch64-darwin`</summary>
<details>
<summary>qdelay</summary>
<pre>    /nix/store/z0dmrxbgqi98f7n8ljza91y5qazdyhcp-clang-wrapper-21.1.8/bin/clang   -arch arm64 -isysroot /nix/store/8pk2m0fn7z5dlb86z5qlv1pxa8w68idx-apple-sdk-14.4/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk -mmacosx-version-min=10.13 -fPIE -MD -MT CMakeFiles/cmTC_b6f4a.dir/testCCompiler.c.o -MF CMakeFiles/cmTC_b6f4a.dir/testCCompiler.c.o.d -o CMakeFiles/cmTC_b6f4a.dir/testCCompiler.c.o -c /nix/build/nix-7082-1392192096/source/build/CMakeFiles/CMakeScratch/TryCompile-levPIh/testCCompiler.c
    Linking C executable cmTC_b6f4a
    /nix/store/3m9pbzqgkm0zijfjjvksh8glz2r95dvi-cmake-4.1.2/bin/cmake -E cmake_link_script CMakeFiles/cmTC_b6f4a.dir/link.txt --verbose=1
    ld: library not found for -lX11
    clang: error: linker command failed with exit code 1 (use -v to see invocation)
    /nix/store/z0dmrxbgqi98f7n8ljza91y5qazdyhcp-clang-wrapper-21.1.8/bin/clang  -arch arm64 -isysroot /nix/store/8pk2m0fn7z5dlb86z5qlv1pxa8w68idx-apple-sdk-14.4/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk -mmacosx-version-min=10.13 -Wl,-search_paths_first -Wl,-headerpad_max_install_names CMakeFiles/cmTC_b6f4a.dir/testCCompiler.c.o -o cmTC_b6f4a
    make[1]: *** [CMakeFiles/cmTC_b6f4a.dir/build.make:103: cmTC_b6f4a] Error 1
    make[1]: Leaving directory &#x27;/nix/build/nix-7082-1392192096/source/build/CMakeFiles/CMakeScratch/TryCompile-levPIh&#x27;
    make: *** [Makefile:134: cmTC_b6f4a/fast] Error 2
    
    



  CMake will not be able to correctly generate this project.
Call Stack (most recent call first):
  CMakeLists.txt:11 (project)


-- Configuring incomplete, errors occurred!</pre>
</details>
</details>

--- a/pkgs/by-name/qd/qdelay/package.nix
+++ b/pkgs/by-name/qd/qdelay/package.nix
@@ -16,6 +16,9 @@
   libXrandr,
   libXtst,
   writableTmpDirAsHomeHook,
+
+  buildVST3 ? true,
+  buildLV2 ? stdenv.isLinux,
 }:
 
 stdenv.mkDerivation (finalAttrs: {
@@ -56,7 +59,10 @@ stdenv.mkDerivation (finalAttrs: {
 
   cmakeFlags = [
     (lib.cmakeBool "COPY_PLUGIN_AFTER_BUILD" false)
-    (lib.cmakeFeature "BUILD_STANDALONE" "OFF")
+
+    (lib.cmakeBool "BUILD_STANDALONE" false)
+    (lib.cmakeBool "BUILD_VST3" buildVST3)
+    (lib.cmakeBool "BUILD_LV2" buildLV2)
   ]
   ++ lib.optionals stdenv.hostPlatform.isDarwin [
     "-DCMAKE_OSX_ARCHITECTURES=${stdenv.hostPlatform.darwinArch}"
@@ -76,10 +82,17 @@ stdenv.mkDerivation (finalAttrs: {
   installPhase = ''
     runHook preInstall
 
-    mkdir -p $out/lib/vst3 $out/lib/lv2
+    pushd QDelay_artefacts/Release
+      ${lib.optionalString buildVST3 ''
+        mkdir -p $out/lib/vst3
+        cp -r VST3/QDelay.vst3 $out/lib/vst3
+      ''}
 
-    cp -r "QDelay_artefacts/Release/LV2/QDelay.lv2" $out/lib/lv2
-    cp -r "QDelay_artefacts/Release/VST3/QDelay.vst3" $out/lib/vst3
+      ${lib.optionalString buildLV2 ''
+        mkdir -p $out/lib/lv2
+        cp -r LV2/QDelay.lv2 $out/lib/lv2
+      ''}
+    popd
 
     runHook postInstall
   '';
@@ -89,7 +102,10 @@ stdenv.mkDerivation (finalAttrs: {
     homepage = "https://github.com/tiagolr/qdelay";
     changelog = "https://github.com/tiagolr/qdelay/releases/tag/v${finalAttrs.version}";
     license = lib.licenses.gpl3Plus;
-    maintainers = with lib.maintainers; [ magnetophon ];
+    maintainers = with lib.maintainers; [
+      magnetophon
+      mrtnvgr
+    ];
     platforms = lib.platforms.all;
   };
 })

